rest/tests/test_rest.py

```diff
@@ -364,15 +365,31 @@ def test_term(self):
 
         # as anonymous
         self.logout()
-        self.get_term(self.repo.slug, vocab_slug, term1['slug'],
+        self.get_term(self.repo.slug, vocab1_slug, term1['slug'],
                       expected_status=HTTP_403_FORBIDDEN)
-        self.get_term(self.repo.slug, vocab_slug, term2['slug'],
+        self.get_term(self.repo.slug, vocab1_slug, term2['slug'],
                       expected_status=HTTP_403_FORBIDDEN)
         self.get_term(self.repo.slug, vocab2_slug, term1['slug'],
                       expected_status=HTTP_403_FORBIDDEN)
         self.get_term(self.repo.slug, vocab2_slug, term2['slug'],
                       expected_status=HTTP_403_FORBIDDEN)
 
+        self.logout()
+        self.login(self.user.username)
+
+        # test delete propagation
+        self.delete_vocabulary(self.repo.slug, vocab1_slug)
+        # vocab1 was deleted and term1 is now also deleted
+        self.get_term(self.repo.slug, vocab1_slug, term1['slug'],
+                      expected_status=HTTP_404_NOT_FOUND)
+        self.get_term(self.repo.slug, vocab1_slug, term2['slug'],
+                      expected_status=HTTP_404_NOT_FOUND,
+                      skip_options_head_test=True)
+        self.get_term(self.repo.slug, vocab2_slug, term1['slug'],
+                      expected_status=HTTP_404_NOT_FOUND,
+                      skip_options_head_test=True)
+        self.get_term(self.repo.slug, vocab2_slug, term2['slug'])
```

Comment:
Maybe a comment saying that this is asserting the other term was not deleted in the cascade (unless I'm misreading the test).  It might be nice to have more descriptive names than vocab1 and vocab2.  Maybe something to describe what makes them different, or descriptive of what they are being used to test.  The test is a bit hard for me to easily follow as is